airbyte-cdk/bulk/core/load/src/main/kotlin/io/airbyte/cdk/load/message/DestinationMessage.kt

```diff
@@ -564,9 +590,18 @@ data class StreamCheckpoint(
         totalBytes = totalBytes
     )
 
+    override fun updateStats(
+        destinationStats: Stats?,
+        totalRecords: Long?,
+        totalBytes: Long?,
+        totalRejectedRecords: Long?
+    ) {
+        destinationStats?.let { this.destinationStats = it }
+        totalRecords?.let { this.totalRecords = it }
+        totalBytes?.let { this.totalBytes = it }
+        totalRejectedRecords?.let { this.totalRejectedRecords = it }
+    }
```

Comment:
Do we rely on these null checks at all or is it just for good measure?

Reply:
yes, not all caller update all. This should maybe be updated, I was trying to expose a `.copy()` that followed the same pattern as the data class copy, but then I figured that it's an object that we should mutate rather than copy.